xsuite/src/data/encoding.ts

```diff
@@ -251,6 +251,16 @@ export const e = {
     }
     return account as PreserveDefinedness<T, Account>;
   },
+  MapperKey: (...args: EncodableMapperKeyArgs) => {
+    const [name, ...vars] = args;
+    return e.Tuple(e.TopStr(name), ...vars);
+  },
+  EsdtRolesKey: (id: string) => e.TopStr(`ELRONDroleesdt${id}`),
```

Comment:
Same as `lastNonce`, I named it the same as the attributes in the object `EncodableEsdt`.
So `lastNonce` instead of `nonce` and `roles` instead of `role` (compared to the blockchain). What do you think?

Reply:
Is "roles" giving the list of things the account can do? E.g. minting, burning, etc. I just don't remember precisely. In that case it makes sense I would say

Reply:
Yes it does